src/parsers/QASM3Parser.cpp

```diff
@@ -726,24 +726,31 @@ class OpenQasm3Parser final : public InstVisitor {
     }
 
     qc::ComparisonKind comparisonKind = qc::ComparisonKind::Eq;
+    qc::ComparisonKind flippedComparisonKind = qc::ComparisonKind::Eq;
     switch (condition->op) {
     case BinaryExpression::Op::LessThan:
       comparisonKind = qc::ComparisonKind::Lt;
+      flippedComparisonKind = qc::ComparisonKind::Geq;
       break;
     case BinaryExpression::Op::LessThanOrEqual:
       comparisonKind = qc::ComparisonKind::Leq;
+      flippedComparisonKind = qc::ComparisonKind::Gt;
       break;
     case BinaryExpression::Op::GreaterThan:
       comparisonKind = qc::ComparisonKind::Gt;
+      flippedComparisonKind = qc::ComparisonKind::Leq;
       break;
     case BinaryExpression::Op::GreaterThanOrEqual:
       comparisonKind = qc::ComparisonKind::Geq;
+      flippedComparisonKind = qc::ComparisonKind::Lt;
       break;
     case BinaryExpression::Op::Equal:
       comparisonKind = qc::ComparisonKind::Eq;
+      flippedComparisonKind = qc::ComparisonKind::Neq;
       break;
     case BinaryExpression::Op::NotEqual:
       comparisonKind = qc::ComparisonKind::Neq;
+      flippedComparisonKind = qc::ComparisonKind::Eq;
       break;
     default:
       error("Unsupported comparison operator.", ifStatement->debugInfo);
```

Comment:
This logic looks like it could be extracted into a function somewhere.
Maybe as a function `getInvertedComparsionKind(const ComparisonKind kind)` somewhere close to the `ComparisonKind` and a method `getComparisonKind(const BinaryExpression::Op op)` somewhere close to the `BinaryExpression`.

Could be a nice separation of concerns.